shared/domains/chelonia/chelonia.js

```diff
@@ -591,25 +591,24 @@ export default (sbp('sbp/selectors/register', {
       sbp('chelonia/private/stopClockSync')
     }
     sbp('chelonia/private/startClockSync')
-    const resyncOnOnline = () => {
-      // Some messages could have been lost between the time the subscription
-      // was requested and it was actually set up. In these cases, force sync
-      // contracts to get them updated.
-      sbp('chelonia/private/out/sync', Array.from(this.subscriptionSet), { force: true }).catch(err => {
-        console.warn(`[chelonia] Syncing contracts failed: ${err.message}`)
-      })
-    }
     this.pubsub = createClient(pubsubURL, {
       ...this.config.connectionOptions,
       handlers: {
         ...options.handlers,
-        online () {
-          resyncOnOnline()
-          options.handlers?.['online']?.()
-        },
-        'reconnection-succeeded' () {
-          resyncOnOnline()
-          options.handlers?.['reconnection-succeeded']?.()
+        // Every time we get a REQUEST_TYPE.SUB response, which happens for
+        // 'new' subscriptions as well as every time the connection is reset
+        'subscription-succeeded': (event) => {
+          const { channelID } = event.detail
+          if (this.subscriptionSet.has(channelID)) {
```

Comment:
Why would we need this `if` check if we're inside of the `'subscription-succeeded'` event? Isn't it guaranteed to be in there? (If not, could you add a comment explaining why it's necessary?)

Reply:
It's not guaranteed to be in there because we could have unsubscribed since.